mobile/ios/ios-device.ts

```diff
@@ -107,8 +126,126 @@ export class IOSDevice implements Mobile.IIOSDevice {
 	}
 
 	private stopSession() {
-			var result = this.$mobileDevice.deviceStopSession(this.devicePointer);
-			this.validateResult(result, "Unable to stop session");
+		var result = this.$mobileDevice.deviceStopSession(this.devicePointer);
+		this.validateResult(result, "Unable to stop session");
+	}
+
+	private getDeviceValue(value: string): string {
+		var deviceCopyValue = this.$mobileDevice.deviceCopyValue(this.devicePointer, null, this.$coreFoundation.createCFString(value));
+		return this.$coreFoundation.convertCFStringToCString(deviceCopyValue);
+	}
+
+	private lookupApplications(): IDictionary<any> {
+		this.connect();
+		try {
```

Comment:
this pattern -> try / startSession / try / finally / stopSession / finally  / disconnect is used quite often (and even mis-used in the getValue function). Should we extract it in a helper method?